### PR TITLE
[release-1.23] Bump images to v1.23.0

### PR DESCRIPTION
**Pull request by dsimansk**
Per title, bump images after release branch cut.

/cc @mgencur @maschmid @skonto

### CODE REVIEW

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -405,7 +405,7 @@ spec:
                 containers:
                   - name: knative-operator
                     # This reference will be replaced in local builds and CI via hack/lib/catalogsource.bash.
-                    image: registry.ci.openshift.org/openshift/openshift-serverless-nightly:openshift-knative-operator
+                    image: registry.ci.openshift.org/openshift/openshift-serverless-v1.23.0:openshift-knative-operator
```
Comment (skonto) [inline olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:408]: These images are not built yet?

**Comment by skonto** [comment]
Regarding bundle and index images we want to keep the nightly ones?

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (dsimansk) [reply olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:408]: It seems that image job failed yesterday. Well, I guess we'll need to trigger CI again first.

**Comment by dsimansk** [comment]
> The changes look ok. Regarding bundle and index images we want to keep the nightly ones?

That's a good point. Probably align to `v1.23.0` as well.

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (dsimansk) [reply olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:408]: That should do the trick. https://github.com/openshift-knative/serverless-operator/pull/1587

Seems that config wasn't available for the first run. I could get it now, so :crossed_fingers:. 

https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-openshift-knative-serverless-operator-release-1.23-4.10-images/1529535682949681152

**Comment by dsimansk** [comment]
@mgencur does it look similar to the "proper CatalogueSource and Tracing" issue that's being addressed in other PR?

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1586/pull-ci-openshift-knative-serverless-operator-release-1.23-4.6-operator-e2e-aws-ocp-46/1530110592038211584#1:build-log.txt%3A1644

```

namespace "knative-serving" deleted
10:10:55.932 INFO:    Ensure no ingress pods running
10:10:55.933 DEBUG:   [[ $(oc get pods -n knative-serving-ingress --field-selector=status.phase!=Succeeded -o jsonpath='{.items}') != '[]' ]] : Waiting until non-zero (max 600 sec.)
done
10:10:56.220 DEBUG:   [[ $(oc get ns knative-serving-ingress --no-headers | wc -l) == 1 ]] : Waiting until non-zero (max 600 sec.)
done
10:10:57.363 INFO:    Removing KnativeEventing CR
knativeeventing.operator.knative.dev "knative-eventing" deleted
10:11:17.863 INFO:    Removing KnativeKafka CR
knativekafka.operator.serverless.openshift.io "knative-kafka" deleted
10:11:20.702 INFO:    Ensure no knative eventing or knative kafka pods running
10:11:20.704 DEBUG:   [[ $(oc get pods -n knative-eventing --field-selector=status.phase!=Succeeded -o jsonpath='{.items}') != '[]' ]] : Waiting until non-zero (max 600 sec.)
.........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
10:24:33.631 ERROR:   Time out of 600 exceeded
10:24:33.632 ERROR:   🚨 Error (code: 71) occurred at ./hack/lib/serverless.bash:23, with command: return 71
10:24:33.634 ERROR:   Stack trace:
10:24:33.635 ERROR:    1: ./hack/teardown.sh:10 teardown_serverless(...)
make: *** [test-e2e-with-kafka] Error 71
{"component":"entrypoint","error":"wrapped process failed: exit status 2","file":"k8s.io/test-infra/prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2022-05-27T10:24:33Z"} 
```

File: templates/csv.yaml [hunk 2]
```diff
@@ -685,12 +685,12 @@ spec:
   relatedImages:
     - name: knative-operator
       # This reference will be replaced in local builds and CI via hack/lib/catalogsource.bash.
-      image: registry.ci.openshift.org/openshift/openshift-serverless-nightly:openshift-knative-operator
+      image: registry.ci.openshift.org/openshift/openshift-serverless-v1.23.0:openshift-knative-operator
     - name: knative-openshift
       # This reference will be replaced in local builds and CI via hack/lib/catalogsource.bash.
-      image: registry.ci.openshift.org/openshift/openshift-serverless-nightly:knative-operator
+      image: registry.ci.openshift.org/openshift/openshift-serverless-v1.23.0:knative-operator
```
Comment (matzew) [inline templates/csv.yaml:691]:
```
docker pull registry.ci.openshift.org/openshift/openshift-serverless-v1.23.0:knative-operator
Error: initializing source docker://registry.ci.openshift.org/openshift/openshift-serverless-v1.23.0:knative-operator: reading manifest knative-operator in registry.ci.openshift.org/openshift/openshift-serverless-v1.23.0: name unknown
```

Comment (matzew) [reply templates/csv.yaml:691]: looks like they are not really there, yet - did the actual version build/promotion happen?

Comment (dsimansk) [reply templates/csv.yaml:691]: Wtf? There was a slight delay in promotion, but I was able to pull all of them locally last week.

Comment (dsimansk) [reply templates/csv.yaml]: Well, maybe that should be better. https://github.com/openshift-knative/serverless-operator/pull/1586/commits/6440ce15b5e42475dd5026299b2d568136072ba2

**Comment by aliok** [comment]
@dsimansk @mgencur 

I couldn't find this in the logs

`delete mutating webhook configuration defaulting.webhook.kafka.messaging.knative.dev: Unauthorized",`

Which job was and run was it, if you still have the browser tab open?

**Comment by mgencur** [comment]
> I couldn't find this in the logs
> 
> `delete mutating webhook configuration defaulting.webhook.kafka.messaging.knative.dev: Unauthorized",`

There was one pod in knative-eventing namespace that was in crash-loop and it was in its logs.

**Comment by dsimansk** [comment]
Here's teardown log: 
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1586/pull-ci-openshift-knative-serverless-operator-release-1.23-4.10-operator-e2e-aws-ocp-410/1531188821079625728/artifacts/operator-e2e-aws-ocp-410/operator-e2e/artifacts/build-4z2b9AGx/debuglog-teardown.sh.log

That has a lot of noise from that failing pod. 


The failing pod itself:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1586/pull-ci-openshift-knative-serverless-operator-release-1.23-4.10-operator-e2e-aws-ocp-410/1531188821079625728/artifacts/operator-e2e-aws-ocp-410/gather-extra/artifacts/pods/knative-eventing_kafka-controller-post-install-1.23.0-829gg_post-install.log

The artifact dir:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1586/pull-ci-openshift-knative-serverless-operator-release-1.23-4.10-operator-e2e-aws-ocp-410/1531188821079625728/artifacts/

It really seems related to the pinned images somehow, rather than race condition during delete etc. It was failing pretty consistently here. However, there's [other PR ](https://github.com/openshift-knative/serverless-operator/pull/1584)to release-1.23 that passed reasonably well.

**Comment by mgencur** [comment]
> [Fix namespace of CI images](https://github.com/openshift-knative/serverless-operator/pull/1586/commits/6440ce15b5e42475dd5026299b2d568136072ba2)

Good catch! I will send the same against "main" because it should be there as well: https://github.com/openshift-knative/serverless-operator/pull/1590

**Comment by mgencur** [comment]
@dsimansk Hey! My theory for the failures is following:
The serverless-stop-bundle 1.22.0 references wrong (outdated) operator images because the namespace in CSV on that branch is wrong as well: https://github.com/openshift-knative/serverless-operator/blob/release-1.22/olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml#L408
I think we need to do the following:
* fix the namespace in that CSV on release-1.22 branch
* rebuild the stop-bundle image on release-1.22 but at this point the images will be promoted to `knative/openshift-serverless-v1.22.1`
* perhaps change https://github.com/openshift-knative/serverless-operator/blob/main/olm-catalog/serverless-operator/project.yaml#L7 to 1.22.1 and regenerate CSV and Dockerfiles

**Comment by mgencur** [comment]
Sent https://github.com/openshift-knative/serverless-operator/pull/1593 to fix the image references on release-1.22 branch.

**Comment by mgencur** [comment]
Sent https://github.com/openshift-knative/serverless-operator/pull/1593 to fix the image references on release-1.22 branch.

**Comment by mgencur** [comment]
Also, trying here if upgrade tests pass on "main" now: https://github.com/openshift-knative/serverless-operator/pull/1592 
IMO, the changes that are currently in this PR for release-1.23 should have no real effect in CI because the images are replaced with those built in CI.

**Comment by mgencur** [comment]
Also, trying here if upgrade tests pass on "main" now: https://github.com/openshift-knative/serverless-operator/pull/1592 
IMO, the changes that are currently in this PR for release-1.23 should have no real effect in CI because the images are replaced with those built in CI.

**Comment by dsimansk** [comment]
@mgencur well at least we moved to get pass the operator tests. It seems like a good direction with your stop bundle theory.

IMO the failed jobs might be still be a flake, given there seems to 3 different kind of failures.

/test 4.6-upstream-e2e-aws-ocp-46

File: hack/lib/scaleup.bash [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  echo -e "\n\n"
+  logger.error "Timeout waiting for scale up to $1 replicas"
+  return 1
+}
+
+function cluster_scalable {
+  if [[ $(oc get infrastructure cluster -ojsonpath='{.status.platform}') = VSphere ]]; then
+    return 1
+  fi
+}
```
Comment (pierDipi) [inline hack/lib/scaleup.bash:65]: Can we use this function https://github.com/openshift-knative/hack/pull/1 ?

Comment (pierDipi) [reply hack/lib/scaleup.bash:65]: or a different one but in a common plane like the hack repository?

**Comment by dsimansk** [comment]
/retest

It seems there's infra issue on 4.6 tests, otherwise upgrade tests seems a bit more stable. 

``` level=info msg="Creating infrastructure resources..."
level=error
level=error msg="Error: [ERR]: Error building changeset: InvalidChangeBatch: [Tried to create resource record set [name='api.ci-op-ww8d6x9h-2b92b.origin-ci-int-aws.dev.rhcloud.com.', type='A'] but it already exists]"
level=error msg="\tstatus code: 400, request id: f401228d-4986-4b78-a23d-f7f254cccf41"
level=error
level=error msg="  on ../tmp/openshift-install-294041980/route53/base.tf line 36, in resource \"aws_route53_record\" \"api_external_alias\":"
level=error msg="  36: resource \"aws_route53_record\" \"api_external_alias\" {"
level=error
level=error
level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply Terraform: failed to complete the change" ```

**Pull request merged**